lib/bambora/v1/reports/merchants.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Bambora
+  module V1
+    module Reports
+      class Merchants
+        def self.get(merchant_id, options = {})
+          credentials = options[:credentials]
+
+          raise ArgumentError, "#{self}.get takes an integer argument" unless merchant_id.is_a?(Integer)
+
+          response =
+            Bambora::Rest::JSONClient
+              .new(base_url: 'https://api.na.bambora.com', merchant_id: credentials.merchant_id)
+              .get(path: "/v1/reports/merchants/#{merchant_id}", api_key: credentials.reporting_passcode)
+
+          raise InvalidAuthenticationError, response if response[:message] == 'Authentication failed'
+
+          raise InvalidRequestError, response if response == { Message: 'The request is invalid.' }
```

Comment:
Here is an example of where we raise the custom errors, to improve developer experience.